engine/src/main/java/com/datatorrent/stram/StreamingContainerParent.java

```diff
@@ -208,20 +212,15 @@ public ContainerHeartbeatResponse processHeartbeat(ContainerHeartbeat msg)
     //LOG.debug("RPC latency from child container {} is {} ms (according to system clocks)", msg.getContainerId(),
     // now - msg.sentTms);
     dagManager.updateRPCLatency(msg.getContainerId(), now - msg.sentTms);
-    try {
-      final ContainerHeartbeat fmsg = msg;
-      return SecureExecutor.execute(new SecureExecutor.WorkLoad<ContainerHeartbeatResponse>()
+    final ContainerHeartbeat fmsg = msg;
+    return SecureExecutor.execute(new SecureExecutor.WorkLoad<ContainerHeartbeatResponse>()
+    {
+      @Override
+      public ContainerHeartbeatResponse run()
       {
-        @Override
-        public ContainerHeartbeatResponse run()
-        {
-          return dagManager.processHeartbeat(fmsg);
-        }
-      });
-    } catch (IOException ex) {
-      LOG.error("Error processing heartbeat", ex);
```

Comment:
Before the exception was swallowed and `null` was returned, causing NullPointerException on the RPC client/container side. It new propagates the exception to the RPC client, so standard mechanism for RPC timeout and recovery is invoked.